contracts/PetPark.sol

```diff
@@ -1,7 +1,87 @@
 //SPDX-License-Identifier: Unlicense
 pragma solidity ^0.8.0;
 
-
 contract PetPark {
+    address public owner;
+
+    constructor() {
+        owner = msg.sender;
+    }
+
+    modifier onlyOwner() {
+        require(owner == msg.sender, "Not an owner");
+        _;
+    }
+
+    modifier isValidAnimal(AnimalType _atype) {
+        require(_atype > AnimalType.INVALID && _atype <= AnimalType.PARROT, "Invalid animal type");
+        _;
+    }
+
+    enum AnimalType {
+        INVALID,
+        FISH,
+        CAT,
+        DOG,
+        RABBIT,
+        PARROT
+    }
+
+    enum Gender {
+        MALE,
+        FEMALE
+    }
+
+    struct Borrower {
+        uint age;
+        Gender gender;
+        AnimalType animalType;
+    }
+
+    mapping(AnimalType => uint) public animalCounts;
+    mapping(address => Borrower) addressToBorrower;
+
+    event Added(AnimalType animalType, uint count);
+    event Borrowed(AnimalType animalType);
+    event Returned(AnimalType animalType);
+
+    function add(AnimalType _atype, uint _count) public onlyOwner isValidAnimal(_atype) {
+        animalCounts[_atype] += _count;
+        emit Added(_atype, _count);
+    }
+
+    function borrow(uint _age, Gender _gender, AnimalType _atype) public isValidAnimal(_atype) {
```

Comment:
This can be marked as an external function

Reply:
I guess then the other 2 can also be external?

Reply:
Yes